### ISSUE TITLE
"Select Exotic" doesn't work with Aeon armor

### ISSUE THREAD

**Issue by Chaemera** [DestinyItemManager/DIM#7334]
### DIM Version

Version 6.93.0 (release), built on 11/28/2021, 9:03:02 PM

### Browser Details

Chrome

### OS Details

Windows 10

### Describe the bug

Using the loadout optimizer, after choosing a set of stats and mods and receiving some suggested loadouts, I went to the "Select Exotic" section and started cycling through available exotics to see how they changed things. Most behaved as expected but when I picked the "Aeon Soul" exotic gloves, the selection was ignored and I was presented with a list of loadouts that did not include it (some even included other exotics). See attached screenshot:

![image](https://user-images.githubusercontent.com/1640327/143885391-08dc763c-a58b-48aa-9516-69fbea295e07.png)

Removing the element-locked gloves mod from my selected list resulted in a "no builds found" list, which at least seems more correct.

### Logs

```shell
chrome-extension://gomekmidlodglbbmalcneegieacbdmki/locales/Locale-en-US.json:1 Failed to load resource: net::ERR_FILE_NOT_FOUND
s-install.avcdn.net/aos/assets/prod/translations/Locale-en.json:1 Failed to load resource: net::ERR_BLOCKED_BY_CLIENT
s-install.avcdn.net/aos/assets/prod/translations/Locale-en-US.json:1 Failed to load resource: net::ERR_BLOCKED_BY_CLIENT
​ Langauge Detected: en-us
​ Langauge Assigned: en
sentry.io/api/279673/envelope/?sentry_key=1367619d45da481b8148dd345c1a1330&sentry_version=7:1 Failed to load resource: net::ERR_BLOCKED_BY_CLIENT
www.google-analytics.com/analytics.js:1 Failed to load resource: net::ERR_BLOCKED_BY_CLIENT
​ [dim sync] globalSettings Object
​ [app] DIM v6.93.0 (release) - Please report any errors to https://www.github.com/DestinyItemManager/DIM/issues
/4611686018468310272/d2/inventory:1 Unchecked runtime.lastError: The message port closed before a response was received.
/4611686018468310272/d2/inventory:1 Error handling response: TypeError: Cannot read properties of undefined (reading 'loggedOut')
    at chrome-extension://kcnhkahnjcbndmmehfkdnkjomaanaooo/contentscript.js:167:137
raw.githubusercontent.com/48klocs/dim-wish-list-sources/master/voltron.txt:1 Failed to load resource: net::ERR_BLOCKED_BY_CLIENT
​ [wishlist] Unable to load wish list TypeError: Failed to fetch
    at 537-e41f2f.js:2
    at main-67624d.js:1
    at Array.map (<anonymous>)
    at main-67624d.js:1
(anonymous) @ 537-e41f2f.js:2
​ Load manifest: 1376.6259765625 ms
​ [dim sync] Saving profile data to IDB
​ Process inventory: 75.670166015625 ms
​ Inventory state update: 334.443115234375 ms
​ Process inventory: 85.004150390625 ms
​ Inventory state update: 73.882080078125 ms
widget_iframe.a53eecb4584348a2ad32ec2ae21f6eae.html:1 Unchecked runtime.lastError: The message port closed before a response was received.
widget_iframe.a53eecb4584348a2ad32ec2ae21f6eae.html:1 Error handling response: TypeError: Cannot read properties of undefined (reading 'loggedOut')
    at chrome-extension://kcnhkahnjcbndmmehfkdnkjomaanaooo/contentscript.js:167:137
​ crbug/1173575, non-JS module files deprecated.
(anonymous) @ chrome-error://chromewebdata/:6770
sentry.io/api/279673/envelope/?sentry_key=1367619d45da481b8148dd345c1a1330&sentry_version=7:1 Failed to load resource: net::ERR_BLOCKED_BY_CLIENT
sentry.io/api/279673/envelope/?sentry_key=1367619d45da481b8148dd345c1a1330&sentry_version=7:1 Failed to load resource: net::ERR_BLOCKED_BY_CLIENT
raw.githubusercontent.com/48klocs/dim-wish-list-sources/master/voltron.txt:1 Failed to load resource: net::ERR_BLOCKED_BY_CLIENT
​ [wishlist] Unable to load wish list TypeError: Failed to fetch
    at 537-e41f2f.js:2
    at main-67624d.js:1
    at Array.map (<anonymous>)
    at main-67624d.js:1
(anonymous) @ 537-e41f2f.js:2
sentry.io/api/279673/envelope/?sentry_key=1367619d45da481b8148dd345c1a1330&sentry_version=7:1 Failed to load resource: net::ERR_BLOCKED_BY_CLIENT
sentry.io/api/279673/envelope/?sentry_key=1367619d45da481b8148dd345c1a1330&sentry_version=7:1 Failed to load resource: net::ERR_BLOCKED_BY_CLIENT
paracausal.science/xur/current.json:1 Failed to load resource: net::ERR_BLOCKED_BY_CLIENT
​ [xur] TypeError: Failed to fetch
    at 537-e41f2f.js:2
    at main-67624d.js:1
    at main-67624d.js:1
    at 657-d482f5.js:2
    at dispatch (657-d482f5.js:2)
    at main-67624d.js:1
(anonymous) @ 537-e41f2f.js:2
sentry.io/api/279673/envelope/?sentry_key=1367619d45da481b8148dd345c1a1330&sentry_version=7:1 Failed to load resource: net::ERR_BLOCKED_BY_CLIENT
sentry.io/api/279673/envelope/?sentry_key=1367619d45da481b8148dd345c1a1330&sentry_version=7:1 Failed to load resource: net::ERR_BLOCKED_BY_CLIENT
sentry.io/api/279673/envelope/?sentry_key=1367619d45da481b8148dd345c1a1330&sentry_version=7:1 Failed to load resource: net::ERR_BLOCKED_BY_CLIENT
sentry.io/api/279673/envelope/?sentry_key=1367619d45da481b8148dd345c1a1330&sentry_version=7:1 Failed to load resource: net::ERR_BLOCKED_BY_CLIENT
9[loadout optimizer] Processing 511056 combinations from 108 items [object Object]
​ [loadout optimizer] found 49998 stat mixes after processing 511056 stat combinations in 283.80000001192093 ms -  1800761 combos/s [object Object] [object Object]
​ [loadout optimizer] useProcess: worker time 322.30000001192093ms
​ [loadout optimizer] useProcess 461.5ms
9[loadout optimizer] Processing 1221480 combinations from 114 items [object Object]
​ [loadout optimizer] found 76343 stat mixes after processing 1221480 stat combinations in 766.6999999880791 ms -  1593165 combos/s [object Object] [object Object]
​ [loadout optimizer] useProcess: worker time 984ms
​ [loadout optimizer] useProcess 1041ms
​ [loadout optimizer] found 49427 stat mixes after processing 1221480 stat combinations in 660.9000000059605 ms -  1848206 combos/s [object Object] [object Object]
​ [loadout optimizer] useProcess: worker time 753ms
​ [loadout optimizer] useProcess 798ms
​ [loadout optimizer] found 25510 stat mixes after processing 1221480 stat combinations in 646.0999999940395 ms -  1890543 combos/s [object Object] [object Object]
​ [loadout optimizer] useProcess: worker time 756.1999999880791ms
​ [loadout optimizer] useProcess 804.9000000059605ms
​ [loadout optimizer] found 3804 stat mixes after processing 1221480 stat combinations in 399.40000000596046 ms -  3058287 combos/s [object Object] [object Object]
​ [loadout optimizer] useProcess: worker time 491.30000001192093ms
​ [loadout optimizer] useProcess 543.4000000059605ms
​ [loadout optimizer] found 155 stat mixes after processing 1221480 stat combinations in 368.59999999403954 ms -  3313836 combos/s [object Object] [object Object]
​ [loadout optimizer] useProcess: worker time 452.90000000596046ms
​ [loadout optimizer] useProcess 500.5ms
​ [loadout optimizer] found 70 stat mixes after processing 1221480 stat combinations in 352.5 ms -  3465191 combos/s [object Object] [object Object]
​ [loadout optimizer] useProcess: worker time 440.5ms
​ [loadout optimizer] useProcess 478.5ms
​ [loadout optimizer] found 26 stat mixes after processing 1221480 stat combinations in 349.40000000596046 ms -  3495935 combos/s [object Object] [object Object]
​ [loadout optimizer] useProcess: worker time 404.80000001192093ms
​ [loadout optimizer] useProcess 455.2999999821186ms
​ [loadout optimizer] Processing 45240 combinations from 72 items [object Object]
​ [loadout optimizer] found 0 stat mixes after processing 45240 stat combinations in 168.19999998807907 ms -  268965 combos/s [object Object] [object Object]
​ [loadout optimizer] useProcess: worker time 351ms
​ [loadout optimizer] useProcess 373.40000000596046ms
5[loadout optimizer] Processing 90480 combinations from 73 items [object Object]
​ [loadout optimizer] found 5 stat mixes after processing 90480 stat combinations in 210.90000000596046 ms -  429018 combos/s [object Object] [object Object]
​ [loadout optimizer] useProcess: worker time 239.5ms
​ [loadout optimizer] useProcess 336.60000002384186ms
​ [loadout optimizer] found 11 stat mixes after processing 90480 stat combinations in 138 ms -  655652 combos/s [object Object] [object Object]
​ [loadout optimizer] useProcess: worker time 197.7000000178814ms
​ [loadout optimizer] useProcess 267.69999998807907ms
8[loadout optimizer] Processing 3120 combinations from 45 items [object Object]
​ [loadout optimizer] found 1 stat mixes after processing 3120 stat combinations in 3.9000000059604645 ms -  799999 combos/s [object Object] [object Object]
​ [loadout optimizer] useProcess: worker time 74.69999998807907ms
​ [loadout optimizer] useProcess 108.80000001192093ms
​ [loadout optimizer] found 1 stat mixes after processing 3120 stat combinations in 4.700000017881393 ms -  663829 combos/s [object Object] [object Object]
​ [loadout optimizer] useProcess: worker time 83.40000000596046ms
​ [loadout optimizer] useProcess 99.59999999403954ms
​ [loadout optimizer] found 1 stat mixes after processing 3120 stat combinations in 3.300000011920929 ms -  945454 combos/s [object Object] [object Object]
​ [loadout optimizer] useProcess: worker time 33.19999998807907ms
​ [loadout optimizer] useProcess 46.099999994039536ms
​ [loadout optimizer] found 1 stat mixes after processing 3120 stat combinations in 3.300000011920929 ms -  945454 combos/s [object Object] [object Object]
​ [loadout optimizer] useProcess: worker time 31.799999982118607ms
​ [loadout optimizer] useProcess 41.79999998211861ms
​ [loadout optimizer] Processing 6240 combinations from 46 items [object Object]
​ [loadout optimizer] found 0 stat mixes after processing 6240 stat combinations in 6 ms -  1040000 combos/s [object Object] [object Object]
​ [loadout optimizer] useProcess: worker time 36.599999994039536ms
​ [loadout optimizer] useProcess 47.69999998807907ms
​ [loadout optimizer] found 5 stat mixes after processing 90480 stat combinations in 185.90000000596046 ms -  486713 combos/s [object Object] [object Object]
​ [loadout optimizer] useProcess: worker time 212.09999999403954ms
​ [loadout optimizer] useProcess 303.19999998807907ms
sentry.io/api/279673/envelope/?sentry_key=1367619d45da481b8148dd345c1a1330&sentry_version=7:1 Failed to load resource: net::ERR_BLOCKED_BY_CLIENT
sentry.io/api/279673/envelope/?sentry_key=1367619d45da481b8148dd345c1a1330&sentry_version=7:1 Failed to load resource: net::ERR_BLOCKED_BY_CLIENT
opencollective.com/dim/banner.js?style={%22h2%22:{%22color%22:%22white%22}}:1 Failed to load resource: net::ERR_BLOCKED_BY_CLIENT
sentry.io/api/279673/envelope/?sentry_key=1367619d45da481b8148dd345c1a1330&sentry_version=7:1 Failed to load resource: net::ERR_BLOCKED_BY_CLIENT
sentry.io/api/279673/envelope/?sentry_key=1367619d45da481b8148dd345c1a1330&sentry_version=7:1 Failed to load resource: net::ERR_BLOCKED_BY_CLIENT
raw.githubusercontent.com/48klocs/dim-wish-list-sources/master/voltron.txt:1 Failed to load resource: net::ERR_BLOCKED_BY_CLIENT
instrument.ts:130 [wishlist] Unable to load wish list TypeError: Failed to fetch
    at instrument.ts:159
    at wishlist-fetch.ts:83
    at Array.map (<anonymous>)
    at wishlist-fetch.ts:82
(anonymous) @ instrument.ts:130
sentry.io/api/279673/envelope/?sentry_key=1367619d45da481b8148dd345c1a1330&sentry_version=7:1 Failed to load resource: net::ERR_BLOCKED_BY_CLIENT
sentry.io/api/279673/envelope/?sentry_key=1367619d45da481b8148dd345c1a1330&sentry_version=7:1 Failed to load resource: net::ERR_BLOCKED_BY_CLIENT
sentry.io/api/279673/envelope/?sentry_key=1367619d45da481b8148dd345c1a1330&sentry_version=7:1 Failed to load resource: net::ERR_BLOCKED_BY_CLIENT
sentry.io/api/279673/envelope/?sentry_key=1367619d45da481b8148dd345c1a1330&sentry_version=7:1 Failed to load resource: net::ERR_BLOCKED_BY_CLIENT
sentry.io/api/279673/envelope/?sentry_key=1367619d45da481b8148dd345c1a1330&sentry_version=7:1 Failed to load resource: net::ERR_BLOCKED_BY_CLIENT
sentry.io/api/279673/envelope/?sentry_key=1367619d45da481b8148dd345c1a1330&sentry_version=7:1 Failed to load resource: net::ERR_BLOCKED_BY_CLIENT
​ [loadout optimizer] found 49998 stat mixes after processing 511056 stat combinations in 279.69999998807907 ms -  1827157 combos/s [object Object] [object Object]
instrument.ts:130 [loadout optimizer] useProcess: worker time 314.80000001192093ms
instrument.ts:130 [loadout optimizer] useProcess 428.69999998807907ms
​ [loadout optimizer] found 34070 stat mixes after processing 511056 stat combinations in 289.90000000596046 ms -  1762869 combos/s [object Object] [object Object]
instrument.ts:130 [loadout optimizer] useProcess: worker time 324.2000000178814ms
instrument.ts:130 [loadout optimizer] useProcess 344.2999999821186ms
​ [loadout optimizer] found 20038 stat mixes after processing 511056 stat combinations in 340.5 ms -  1500898 combos/s [object Object] [object Object]
instrument.ts:130 [loadout optimizer] useProcess: worker time 410.39999997615814ms
instrument.ts:130 [loadout optimizer] useProcess 434.40000000596046ms
​ [loadout optimizer] found 3692 stat mixes after processing 511056 stat combinations in 211.69999998807907 ms -  2414057 combos/s [object Object] [object Object]
instrument.ts:130 [loadout optimizer] useProcess: worker time 251.90000000596046ms
instrument.ts:130 [loadout optimizer] useProcess 288.30000001192093ms
​ [loadout optimizer] found 6 stat mixes after processing 511056 stat combinations in 195.09999999403954 ms -  2619456 combos/s [object Object] [object Object]
instrument.ts:130 [loadout optimizer] useProcess: worker time 256.7000000178814ms
instrument.ts:130 [loadout optimizer] useProcess 296.80000001192093ms
​ [loadout optimizer] found 184 stat mixes after processing 511056 stat combinations in 212.19999998807907 ms -  2408369 combos/s [object Object] [object Object]
instrument.ts:130 [loadout optimizer] useProcess: worker time 249.10000002384186ms
instrument.ts:130 [loadout optimizer] useProcess 287.40000000596046ms
​ [loadout optimizer] found 0 stat mixes after processing 511056 stat combinations in 159.19999998807907 ms -  3210150 combos/s [object Object] [object Object]
instrument.ts:130 [loadout optimizer] useProcess: worker time 196.59999999403954ms
instrument.ts:130 [loadout optimizer] useProcess 212.19999998807907ms
​ [loadout optimizer] found 0 stat mixes after processing 511056 stat combinations in 159.69999998807907 ms -  3200100 combos/s [object Object] [object Object]
instrument.ts:130 [loadout optimizer] useProcess: worker time 190.09999999403954ms
instrument.ts:130 [loadout optimizer] useProcess 203.90000000596046ms
​ [loadout optimizer] found 26 stat mixes after processing 1221480 stat combinations in 328.5 ms -  3718356 combos/s [object Object] [object Object]
instrument.ts:130 [loadout optimizer] useProcess: worker time 361.59999999403954ms
instrument.ts:130 [loadout optimizer] useProcess 477.2999999821186ms
​ [loadout optimizer] found 26 stat mixes after processing 1221480 stat combinations in 307.2999999821186 ms -  3974877 combos/s [object Object] [object Object]
instrument.ts:130 [loadout optimizer] useProcess: worker time 370ms
instrument.ts:130 [loadout optimizer] useProcess 421.59999999403954ms
​ [loadout optimizer] found 5 stat mixes after processing 90480 stat combinations in 218.2999999821186 ms -  414475 combos/s [object Object] [object Object]
instrument.ts:130 [loadout optimizer] useProcess: worker time 375.19999998807907ms
instrument.ts:130 [loadout optimizer] useProcess 414.2000000178814ms
​ [loadout optimizer] found 1 stat mixes after processing 3120 stat combinations in 4.200000017881393 ms -  742857 combos/s [object Object] [object Object]
instrument.ts:130 [loadout optimizer] useProcess: worker time 56.400000005960464ms
instrument.ts:130 [loadout optimizer] useProcess 83ms
​ [loadout optimizer] Processing 6240 combinations from 46 items [object Object]
​ [loadout optimizer] found 0 stat mixes after processing 6240 stat combinations in 8.400000005960464 ms -  742857 combos/s [object Object] [object Object]
instrument.ts:130 [loadout optimizer] useProcess: worker time 40.39999997615814ms
instrument.ts:130 [loadout optimizer] useProcess 49.099999994039536ms
​ [loadout optimizer] found 0 stat mixes after processing 3120 stat combinations in 4.4000000059604645 ms -  709090 combos/s [object Object] [object Object]
instrument.ts:130 [loadout optimizer] useProcess: worker time 82ms
instrument.ts:130 [loadout optimizer] useProcess 92ms
​ [loadout optimizer] found 0 stat mixes after processing 3120 stat combinations in 4.199999988079071 ms -  742857 combos/s [object Object] [object Object]
instrument.ts:130 [loadout optimizer] useProcess: worker time 31ms
instrument.ts:130 [loadout optimizer] useProcess 40.599999994039536ms
​ [loadout optimizer] found 0 stat mixes after processing 3120 stat combinations in 4.199999988079071 ms -  742857 combos/s [object Object] [object Object]
instrument.ts:130 [loadout optimizer] useProcess: worker time 32.400000005960464ms
instrument.ts:130 [loadout optimizer] useProcess 41.20000001788139ms
​ [loadout optimizer] found 5 stat mixes after processing 90480 stat combinations in 208.59999999403954 ms -  433748 combos/s [object Object] [object Object]
instrument.ts:130 [loadout optimizer] useProcess: worker time 275.90000000596046ms
instrument.ts:130 [loadout optimizer] useProcess 362.5ms
​ [loadout optimizer] Processing 9048 combinations from 64 items [object Object]
​ [loadout optimizer] found 0 stat mixes after processing 9048 stat combinations in 7.4000000059604645 ms -  1222702 combos/s [object Object] [object Object]
instrument.ts:130 [loadout optimizer] useProcess: worker time 63.900000005960464ms
instrument.ts:130 [loadout optimizer] useProcess 78.59999999403954ms
```

**Comment by ryan-rushton** [comment]
It is working fine for me. @Chaemera can you confirm for me that you have an Armour 2.0 version of those gloves in your inventory?

![Screen Shot 2021-12-01 at 9 05 24 am](https://user-images.githubusercontent.com/7344652/144135597-9b9c83f6-4bf6-41bf-8dec-8c9de3a0eb8a.png)

**Comment by Chaemera** [comment]
Can confirm, current pair is completely up to date: 
![image](https://user-images.githubusercontent.com/1640327/144140941-e3d8dc31-2656-4fe0-9779-970eb937bfd4.png)

Upon further testing, this seems to occur if I choose an element-locked mod for the gear slot in question that does not match an element available on the selected exotic, tested on other exotics besides the Aeon gloves.
![image](https://user-images.githubusercontent.com/1640327/144141620-83f739dd-fa82-44c5-837c-d37a2ade45cf.png)

Basically, it bypasses the exotic choice to force a fit for the chosen mod.

**Comment by ryan-rushton** [comment]
Yes that is expected at this current point in time. If you were to choose a different armour upgrade option where you can change the element on the armour you would see sets with the given exotic.

If we have a filter that returns no items we purposely just return all the items so something shows up. What we need to do is have some kind of indicator that we are showing all items for said item slot. We have talked about this a few times but it keeps getting put on the TODO list (for me anyways).

**Comment by Chaemera** [comment]
Ah okay. I would've just expected it to return a "no sets found" result, is why I was confused.

**Comment by bhollis** [comment]
I think it should if you specified an exotic - that's different than a search.

**Comment by ryan-rushton** [comment]
Ok will raise something to change it to that.
